server/src/main/java/org/apache/druid/discovery/BaseNodeRoleWatcher.java

```diff
@@ -209,13 +217,52 @@ private void removeNode(DiscoveryDruidNode druidNode)
     }
   }
 
+  public void cacheInitializedTimedOut()
+  {
+    synchronized (lock) {
+      // No need to wait on CountDownLatch, because we are holding the lock under which it could only be
+      // counted down.
+      if (cacheInitialized.getCount() == 0) {
+        LOGGER.warn("cache is already initialized. ignoring timeout.");
```

Comment:
Please update this log message (and all others that don't include it) to include the role, like: `Cache for role[%s] is already initialized. Ignoring timeout.`

Reply:
Done